myhomework5.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+require_once 'HttpStatusCodes.php';
+require_once 'ParserRequest.php';
+const HOST_BASE_DIRECTORIES = [
+    'student.shpp.me' => 'student',
+    'another.shpp.me' => 'another'
+];
+const MY_DIRECTORY = '/Applications/XAMPP/xamppfiles/htdocs/1levelV2/';
+
+//function readHttpLikeInput()
+//{
+//    $f = fopen('php://stdin', 'r');
+//    $store = "";
+//    $toread = 0;
+//    while ($line = fgets($f)) {
+//        $store .= preg_replace("/\r/", "", $line);
+//        if (preg_match('/Content-Length: (\d+)/', $line, $m))
+//            $toread = $m[1] * 1;
+//        if ($line == "\r\n")
+//            break;
+//    }
+//    if ($toread > 0)
+//        $store .= fread($f, $toread);
+//    return $store;
+//}
+//
+//$contents = readHttpLikeInput();
+
+function outputHttpResponse($statuscode, $statusmessage, $headers, $body)
+{
+    $response = "HTTP/1.1 $statuscode $statusmessage" . PHP_EOL;
+    $response .= "Date: " . date("l, d F  Y h:i:sa") . PHP_EOL;
+    $response .= "Server: Apache/2.2.14 (Win32)" . PHP_EOL;
+    $response .= "Content-Length: " . strlen($body) . PHP_EOL;
+    $response .= "Connection: Closed" . PHP_EOL;
+    $response .= "Content-Type: text/html; charset=utf-8" . PHP_EOL;
+    $response .= PHP_EOL . "$body";
+
+    echo $response;
+}
+
+function processHttpRequest($method, $uri, $headers, $body)
+{
+
+    try {
+        $nameFolder = getNameFolder($headers);
+
+        if (checkHost($nameFolder)) {
```

Comment:
I believe you will get an error in `getNameFolder` if host is not found, so no sense in checking host